cachi2/core/package_managers/yarn/resolver.py

```diff
@@ -287,9 +287,11 @@ def _generate_purl_for_package(self, package: _ResolvedPackage, project: Project
         qualifiers = dict()
         subpath = None
 
-        # package with NpmLocator doesn't need any qualifiers
+        if isinstance(package.locator, NpmLocator):
```

Comment:
This change should have been part of the commit which implements the guts of the method: https://github.com/containerbuildsystem/cachi2/pull/374/commits/0709ff1cdb317fbcf363637028842beca45e1341

Reply:
I meant the whole hunk, not just the single line ;) .